### PR TITLE
refactor(datatable): increase code style consistency

### PR DESCRIPTION
**Pull request by alexandereneroth**
## Describe pull-request

This PR:

- Makes it so data table uses the tablePropsChangedEvent for communicating all changes to table props.
- Makes it so child components of data table reads table props on initial load.
- Reorders class member variables and methods for readability and consistency.
- Removed commented code
- Fixes eslint errors

## How to test

Checkout the branch and test in storybook locally
Or
Look at netlify deploy preview

### CODE REVIEW

File: tegel/src/components/data-table/table-footer/table-footer.tsx [hunk 1]
```diff
@@ -102,19 +98,17 @@ export class TableFooter {
...
-    }
-  }
-
   connectedCallback() {
-    this.uniqueTableIdentifier = this.host.closest('sdds-table').getAttribute('id');
+    this.tableEl = this.host.closest('sdds-table');
   }
 
   componentWillLoad() {
+    this.uniqueTableIdentifier = this.host.closest('sdds-table').getAttribute('id');
```
Comment (mJarsater) [inline tegel/src/components/data-table/table-footer/table-footer.tsx:106]: Can't this be: this.uniqueTableIdentifier = this.tableEl.getAttribute('id'); ?

File: tegel/src/components/data-table/table-header-cell/table-header-cell.tsx [hunk 2]
```diff
@@ -176,15 +134,50 @@ export class TableHeaderCell {
...
-    if (this.uniqueTableIdentifier === event.detail[0]) this.enableMultiselect = event.detail[1];
+    if (this.uniqueTableIdentifier === event.detail[0]) [, this.enableMultiselect] = event.detail;
+  }
+
+  connectedCallback() {
+    this.tableEl = this.host.closest('sdds-table');
+  }
+
+  componentWillLoad() {
+    this.uniqueTableIdentifier = this.host.closest('sdds-table').getAttribute('id');
```
Comment (mJarsater) [inline tegel/src/components/data-table/table-header-cell/table-header-cell.tsx:147]: Can't this be: this.uniqueTableIdentifier = this.tableEl.getAttribute('id'); ?

File: tegel/src/components/data-table/table-header/table-header.tsx [hunk 3]
```diff
@@ -133,6 +96,28 @@ export class TableHeaderRow {
     }
   }
 
+  connectedCallback() {
+    this.tableEl = this.host.closest('sdds-table');
+  }
+
+  componentWillLoad() {
+    this.uniqueTableIdentifier = this.host.closest('sdds-table').getAttribute('id');
```
Comment (mJarsater) [inline tegel/src/components/data-table/table-header/table-header.tsx:104]: Can't this be: this.uniqueTableIdentifier = this.tableEl.getAttribute('id'); ?

File: tegel/src/components/data-table/table-toolbar/table-toolbar.tsx [hunk 4]
```diff
@@ -79,11 +67,28 @@ export class TableToolbar {
...
-  commonTableStyleListener(event: CustomEvent<any>) {
-    if (this.uniqueTableIdentifier === event.detail[0]) {
-      [, this.verticalDividers, this.compactDesign, this.noMinWidth, this.whiteBackground] =
-        event.detail;
+  connectedCallback() {
+    this.tableEl = this.host.closest('sdds-table');
+  }
+
+  componentWillLoad() {
+    this.uniqueTableIdentifier = this.host.closest('sdds-table').getAttribute('id');
```
Comment (mJarsater) [inline tegel/src/components/data-table/table-toolbar/table-toolbar.tsx:75]: Can't this be: this.uniqueTableIdentifier = this.tableEl.getAttribute('id'); ?

File: tegel/src/components/data-table/table-body-cell/table-body-cell.tsx [hunk 5]
```diff
@@ -93,6 +76,18 @@ export class TableBodyCell {
     }
   }
 
+  connectedCallback() {
+    this.tableEl = this.host.closest('sdds-table');
+  }
+
+  componentWillLoad() {
+    this.uniqueTableIdentifier = this.host.closest('sdds-table').getAttribute('id');
```
Comment (mJarsater) [inline tegel/src/components/data-table/table-body-cell/table-body-cell.tsx:84]:
```suggestion
    this.uniqueTableIdentifier = this.tableEl.getAttribute('id');
```

File: tegel/src/components/data-table/table-body-row-expandable/table-body-row-expandable.tsx [hunk 6]
```diff
@@ -62,8 +76,20 @@ export class TableBodyRowExpandable {
     }
   }
 
+  connectedCallback() {
+    this.tableEl = this.host.closest('sdds-table');
+  }
+
   componentWillLoad() {
     this.uniqueTableIdentifier = this.host.closest('sdds-table').getAttribute('id');
```
Comment (mJarsater) [inline tegel/src/components/data-table/table-body-row-expandable/table-body-row-expandable.tsx:84]:
```suggestion
    this.uniqueTableIdentifier = this.tableEl.getAttribute('id');
```

File: tegel/src/components/data-table/table-body-row/table-body-row.tsx [hunk 7]
```diff
@@ -108,16 +104,20 @@ export class TableBodyRow {
...
-    } else {
-      row.classList.remove('sdds-table__row--selected');
-    }
-    this.bodyRowToTable.emit(this.bodyCheckBoxStatus);
+  connectedCallback() {
+    this.tableEl = this.host.closest('sdds-table');
+  }
+
+  componentWillLoad() {
+    this.uniqueTableIdentifier = this.host.closest('sdds-table').getAttribute('id');
```
Comment (mJarsater) [inline tegel/src/components/data-table/table-body-row/table-body-row.tsx:112]:
```suggestion
    this.uniqueTableIdentifier = this.tableEl.getAttribute('id');
```

File: tegel/src/components/data-table/table-body/table-body.tsx [hunk 8]
```diff
@@ -373,6 +298,42 @@ export class TableBody {
...
+      this.searchFunction(event.detail[1]);
+    }
+  }
+
+  connectedCallback() {
+    this.tableEl = this.host.closest('sdds-table');
+  }
+
+  componentWillLoad() {
+    this.uniqueTableIdentifier = this.host.closest('sdds-table').getAttribute('id');
```
[the remark this answers is not in the document]
Comment (alexandereneroth) [reply tegel/src/components/data-table/table-body/table-body.tsx]: nice finds

**Review by mJarsater — APPROVED** [review]
LGTM

**Pull request merged**